fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/DistinctRecordsProperty.java

```diff
@@ -90,6 +94,11 @@ public String toString() {
      */
     @API(API.Status.EXPERIMENTAL)
     public static class DistinctRecordsVisitor implements RecordQueryPlanVisitor<Boolean> {
+        @Nonnull
+        @Override
+        public Boolean visitUpdatePlan(@Nonnull final RecordQueryUpdatePlan element) {
+            return true;
```

Comment:
should be probably distinctness of child.

Reply:
> If the input isn't distinct, non-idempotent updates seem problematic.

Is it implementation-specific problem (i.e. Halloween)? or is there more to it? because I think a case like this, although unusual, is allowed, e.g. as per PostgreSQL:

```
postgres=# create table t1(col1 int primary key, col2 int);
CREATE TABLE
postgres=# insert into t1 values(1,1);
INSERT 0 1
postgres=# update t1 set col1=s.col1, col2=s.col2 from (select * from (values (1,5), (1,5), (1,5)) as t(col1, col2)) as s where t1.col1 = s.col1;
UPDATE 1
postgres=# select * from t1;
 col1 | col2
------+------
    1 |    5
(1 row)
```